lib/ts/authUtils.ts

```diff
@@ -934,16 +922,41 @@ async function filterOutInvalidSecondFactorsOrThrowIfAllAreInvalid(
     const mfaInstance = MultiFactorAuthRecipe.getInstance();
     if (mfaInstance !== undefined) {
         if (!inputUserAlreadyLinkedToSessionUser) {
+            let factorsSetUpForUserProm: Promise<string[]> | undefined;
+            let mfaInfoProm: ReturnType<typeof updateAndGetMFARelatedInfoInSession> | undefined;
+            const memoizedAllowedToSetupFactorInput = {
+                factorId: "placeholder", // This is updated inside the loop
+                session,
+
+                get factorsSetUpForUser() {
```

Comment:
so users will have to call this function from their override if they want to get the value of this? But then the type fo the input is `Promise<..>` and not `() => Promise<...>`. So im a bit confused as to how this works